crates/sui-types/src/waypoint.rs

```diff
@@ -0,0 +1,426 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use serde::{Deserialize, Serialize};
+use std::borrow::Borrow;
+use std::fmt::Debug;
+use thiserror::Error;
+
+use std::collections::{BTreeMap, BTreeSet};
+
+use curve25519_dalek::ristretto::RistrettoPoint;
+use ed25519_dalek::Sha512;
+
+use crate::{base_types::AuthorityName, committee::Committee};
+
+#[cfg(test)]
+#[path = "unit_tests/waypoint_tests.rs"]
+mod waypoint_tests;
+
+#[derive(Eq, PartialEq, Clone, Debug, Error)]
+#[allow(clippy::large_enum_variant)]
+pub enum WaypointError {
+    #[error("Waypoint error: {:?}", msg)]
+    Generic { msg: String },
+    #[error("Nothing to do, all parts are already in the checkpoint.")]
+    NothingToDo,
+    #[error("Fragment cannot connect with global checkpoint.")]
+    CannotConnect,
+}
+
+impl WaypointError {
+    pub fn generic(msg: String) -> WaypointError {
+        WaypointError::Generic { msg }
+    }
+}
+
+/*
+   A MulHash accumulator: each element is mapped to a
+   point on an eliptic curve on which the DL problem is
+   hard. The accumulator is the sum of all points.
+*/
+#[derive(Default, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub struct Accumulator {
+    accumulator: RistrettoPoint,
+}
+
+impl Accumulator {
+    /// Insert one item in the accumulator
+    pub fn insert<I>(&mut self, item: &I)
+    where
+        I: Borrow<[u8]>,
```

Comment:
I think the more appropriate trait to use here is `AsRef<[u8]>`, see [book](http://web.mit.edu/rust-lang_v1.25/arch/amd64_ubuntu1404/share/doc/rust/html/book/first-edition/borrow-and-asref.html)

Reply:
Many thanks -- changed!